tests/helm/common_test.go

```diff
@@ -413,6 +413,48 @@ func GetPodSpec(object unstructured.Unstructured) (*corev1.PodSpec, error) {
 	}
 }
 
+func GetPodTemplateSpec(object unstructured.Unstructured) (*corev1.PodTemplateSpec, error) {
+	switch object.GetKind() {
+	case "Deployment":
+		deployment := &appsv1.Deployment{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.Object, deployment)
+		if err != nil {
+			return nil, err
+		}
+		return &deployment.Spec.Template, nil
+	case "StatefulSet":
+		statefulset := &appsv1.StatefulSet{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.Object, statefulset)
+		if err != nil {
+			return nil, err
+		}
+		return &statefulset.Spec.Template, nil
+	case "DaemonSet":
+		daemonset := &appsv1.DaemonSet{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.Object, daemonset)
+		if err != nil {
+			return nil, err
+		}
+		return &daemonset.Spec.Template, nil
+	case "Job":
+		job := &batchv1.Job{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.Object, job)
+		if err != nil {
+			return nil, err
+		}
+		return &job.Spec.Template, nil
+	case "CronJob":
+		cronJob := &batchv1.CronJob{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.Object, cronJob)
+		if err != nil {
+			return nil, err
+		}
+		return &cronJob.Spec.JobTemplate.Spec.Template, nil
+	default:
+		return nil, nil
+	}
+}
+
```

Comment:
It can be simplified to something like this:

```suggestion
func GetPodTemplateSpec(object unstructured.Unstructured) (*corev1.PodTemplateSpec, error) {
	spec, err := GetPodSpec(object)
	if err != nil {
		return nil, err
	}
	return spec.Template, nil
}

```